aten/src/ATen/function_wrapper.py

```diff
@@ -772,7 +772,7 @@ def emit_nn_body(option):
             # _out variants must create buffers and insert them in the
             # arguments list between output and input arguments
             for buffer in option['buffers']:
-                body.append('Tensor {} = at::empty({{0}}, this->options());'.format(buffer['name']))
+                body.append('Tensor {} = at::empty({{0}}, output.options());'.format(buffer['name']))
```

Comment:
I think this works, but I'm not sure if it's guaranteed going forward that the name of the output parameter is "output" -- for consistency it makes the most sense to have this be the name of the tensor that was device_guarded before this call.  Is it feasible to just pass that into this function? (emit_nn_body).